pkg/chains/storage/archivista/archivista.go

```diff
@@ -0,0 +1,217 @@
+// in one single patch call.
+func PatchTektonObjectAnnotations(ctx context.Context, obj objects.TektonObject, annotations map[string]string, tektonClient tektonclient.Interface) error {
+	patchData := map[string]interface{}{
+		"metadata": map[string]interface{}{
+			"annotations": annotations,
+		},
+	}
+	patchBytes, err := json.Marshal(patchData)
+	if err != nil {
+		return fmt.Errorf("failed to marshal patch data: %w", err)
+	}
+
+	switch o := obj.GetObject().(type) {
+	case *tektonv1.TaskRun:
+		_, err = tektonClient.TektonV1().TaskRuns(o.Namespace).Patch(ctx, o.Name, types.MergePatchType, patchBytes, metav1.PatchOptions{})
+		return err
+	case *tektonv1.PipelineRun:
+		_, err = tektonClient.TektonV1().PipelineRuns(o.Namespace).Patch(ctx, o.Name, types.MergePatchType, patchBytes, metav1.PatchOptions{})
+		return err
+	case *v1beta1.TaskRun:
+		_, err = tektonClient.TektonV1beta1().TaskRuns(o.Namespace).Patch(ctx, o.Name, types.MergePatchType, patchBytes, metav1.PatchOptions{})
+		return err
+	case *v1beta1.PipelineRun:
+		_, err = tektonClient.TektonV1beta1().PipelineRuns(o.Namespace).Patch(ctx, o.Name, types.MergePatchType, patchBytes, metav1.PatchOptions{})
+		return err
+	default:
+		return fmt.Errorf("unsupported Tekton object type for patching")
+	}
+}
+
+// StorePayload builds a DSSE envelope from the raw payload and signature,
+// logs the envelope, uploads it via the Archivista client API, and patches the
+// Tekton object with the returned gitoid and Archivista URL.
+func (a *ArchivistaStorage) StorePayload(ctx context.Context, obj objects.TektonObject, rawPayload []byte, signature string, opts config.StorageOpts) error {
+	logger := logging.FromContext(ctx)
+
+	// Validate signature.
+	if strings.TrimSpace(signature) == "" {
+		return fmt.Errorf("missing signature")
+	}
+
+	var keyID string
+	certPEM := strings.TrimSpace(opts.Cert)
+	if certPEM != "" {
+		block, _ := pem.Decode([]byte(certPEM))
+		if block != nil {
+			cert, err := x509.ParseCertificate(block.Bytes)
+			if err == nil {
+				// Generate keyID from the public key.
+				keyID, err = generatePublicKeyIDFunc(cert.PublicKey, crypto.SHA256)
+				if err != nil {
+					logger.Errorw("Failed to generate KeyID", "error", err)
+					keyID = ""
+				}
+			} else {
+				logger.Errorw("Failed to parse certificate", "error", err)
+			}
+		} else {
+			logger.Error("Failed to decode certificate PEM")
+		}
+	} // if no certificate provided, keyID remains blank
+
+	// Optionally decode the payload for logging.
+	decodedPayload, err := base64.StdEncoding.DecodeString(string(rawPayload))
+	if err != nil {
+		logger.Errorw("Failed to base64 decode payload", "keyID", keyID, "error", err)
+		logger.Infof("Raw payload (not base64 decoded): %s", string(rawPayload))
+	} else {
+		logger.Infof("Decoded payload: %s", string(decodedPayload))
+	}
+
+	env := buildEnvelope(rawPayload, signature, keyID, opts.Chain)
```

Comment:
I'm looking into this locally today but I don't think we need all of the dsse construction code and cert code -- chains already has this. Usage here: https://github.com/testifysec/chains/blob/7824cdc5b20e1bffb05453e7360fbfbc5ea45640/pkg/chains/storage/oci/attestation.go#L73